doc/cmake.rst

```diff
@@ -38,6 +38,27 @@ the notable differences.
     Does not define NDEBUG when disabled, NDEBUG is set via Release-like
     CMake build configurations; Release, MinSizeRel.
 
+Testing Integration
+-------------------
+When consuming a third party CMake project into yours, an unfortunate side
+effect is that the targets of the consumed project appear in the list of
+available targets, and are by default included in the ALL meta target
+created by most build systems. For this reason, all the targets specified
+in godot-cpp are marked with the ``EXCLUDE_FROM_ALL`` tag to prevent
+unnecessary compilation. The testing targets ``godot-cpp.test.<target>``
+are also guarded by ``GODOT_ENABLE_TESTING`` which is off by default.
+
+To configure and build the godot-cpp project to enable the integration
+testing targets the command will look something like:
+
+.. code-block::
+
+    # Assuming our current directory is the godot-cpp source root
+    mkdir cmake-build
+    cd cmake-build
+    cmake .. -DGODOT_ENABLE_TESTING=YES
+    cmake --build . --target godot-cpp.test.template_release
```

Comment:
Could the example use `template_debug`? I'd expect almost everyone working on Godot/godot-cpp will be testing with an editor build of Godot, and so would want to make a `template_debug` build

```suggestion
    cmake --build . --target godot-cpp.test.template_debug
```

Reply:
Hm, and maybe we'd want to use `-DCMAKE_BUILD_TYPE=Debug` and `--config Debug` too? Most folks would also want debug symbols

Reply:
done